ofctrl/ofSwitch.go

```diff
@@ -165,40 +153,41 @@ func (self *OFSwitch) IsReady() bool {
 }
 
 // Handle switch connected event
-func (self *OFSwitch) switchConnected() {
-	self.changeStatus(true)
-
-	// Send new feature request
-	self.Send(openflow15.NewFeaturesRequest())
-
-	self.Send(openflow15.NewEchoRequest())
-
-	self.heartbeatCh = make(chan struct{})
+func (self *OFSwitch) switchConnected() error {
+	// Main receive loop for the switch
+	go self.receive()
+	// Periodically sends echo request message on the connection.
 	go func() {
 		timer := time.NewTicker(heartbeatInterval)
 		defer timer.Stop()
 		for {
 			select {
 			case <-timer.C:
-				self.Send(openflow15.NewEchoRequest())
-			case <-self.heartbeatCh:
-				break
+				if err := self.Send(openflow15.NewEchoRequest()); err != nil {
+					log.Errorf("Failed to send echo request, and retry after 3s: %v", err)
```

Comment:
the patch updates the internal to 5s.

Reply:
updated with variable `heartbeatInterval` to avoid hard-code error